Upstream: yes
Reason: Fix build with boost[>=1.66.0]

From c9b5b13fb81b1ed142dee0e843101c6b8832ca95 Mon Sep 17 00:00:00 2001
From: D Benage <dbenage.cx@gmail.com>
Date: Wed, 13 Dec 2017 18:53:55 -0600
Subject: [PATCH] Replace use of boost::system::posix_error

First deprecated in boost v1.37
---
 GG/src/dialogs/FileDlg.cpp | 2 +-
 UI/ClientUI.cpp            | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/GG/src/dialogs/FileDlg.cpp b/GG/src/dialogs/FileDlg.cpp
index 9032aee58..17b923233 100644
--- a/GG/src/dialogs/FileDlg.cpp
+++ b/GG/src/dialogs/FileDlg.cpp
@@ -746,7 +746,7 @@ void FileDlg::OpenDirectory()
             try {
                 SetWorkingDirectory(fs::path(directory + "\\"));
             } catch (const fs::filesystem_error& e) {
-                if (e.code() == boost::system::posix_error::io_error) {
+                if (e.code() == boost::system::errc::io_error) {
                     m_in_win32_drive_selection = true;
                     m_files_edit->Clear();
                     FilesEditChanged(m_files_edit->Text());
diff --git a/UI/ClientUI.cpp b/UI/ClientUI.cpp
index f294e9116..f9709ad4e 100644
--- a/UI/ClientUI.cpp
+++ b/UI/ClientUI.cpp
@@ -1136,7 +1136,7 @@ ClientUI::TexturesAndDist ClientUI::PrefixedTexturesAndDist(const boost::filesys
                     textures.push_back(ClientUI::GetTexture(*it, mipmap));
             } catch (const fs::filesystem_error& e) {
                 // ignore files for which permission is denied, and rethrow other exceptions
-                if (e.code() != boost::system::posix_error::permission_denied)
+                if (e.code() != boost::system::errc::permission_denied)
                     throw;
             }
         }
-- 
2.17.0

